datapackage_pipelines/lib/sort.py

```diff
@@ -20,8 +20,8 @@ def __call__(self, row):
 
 def sorter(resource):
     db = KVStore()
-    for row in resource:
-        key = key_calc(row)
+    for row_num, row in enumerate(resource):
+        key = key_calc(row) + "-{}".format(row_num)
```

Comment:
in a the scenario where rows 999 and 1000 have the same key, row '1000' will come before '999'

Reply:
using `{:08x}` instead could do the trick

Reply:
do you have a good solution for this? I can add some padding, but I think it unnecessarily complicates things.

I'm don't think guaranteeing that rows with duplicate keys are sorted within the key is a priority

maybe it's better to just say that rows with duplicate keys are unsorted (within the duplicated key)?

Reply:
@OriHoch ^ proposed solution above

Reply:
@akariv fixed